crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -177,97 +203,83 @@ where
             );
         };
 
-        // Return the internal result
-        let result = result.unwrap();
-        // minimum_time.await;
-        (peer_name, result)
+        (peer_name, result.unwrap())
     }
 
-    async fn gossip_timeout(&mut self, duration: Duration) -> Result<(), SuiError> {
+    async fn peer_gossip_for_duration(&mut self, duration: Duration) -> Result<(), SuiError> {
         // Global timeout, we do not exceed this time in this task.
         let mut timeout = Box::pin(tokio::time::sleep(duration));
-        let mut queue = FuturesOrdered::new();
 
         let req = BatchInfoRequest {
             start: self.max_seq,
             length: REQUEST_FOLLOW_NUM_DIGESTS,
         };
 
-        // Get a client
         let mut streamx = Box::pin(self.client.handle_batch_stream(req).await?);
 
         loop {
             tokio::select! {
                 _ = &mut timeout => {
-                    // No matter what happens we do not spend too much time
-                    // for any peer.
-
-                    break },
+                    // No matter what happens we do not spend too much time on any peer.
+                    break;
+                },
 
                 items = &mut streamx.next() => {
                     match items {
-                        // Upon receiving a batch
-                        Some(Ok(BatchInfoResponseItem(UpdateItem::Batch(_signed_batch)) )) => {
-                            // Update the longer term sequence_number only after a batch that is signed
-                            self.max_seq = Some(_signed_batch.batch.next_sequence_number);
-                        },
-                        // Upon receiving a transaction digest we store it, if it is not processed already.
-                        Some(Ok(BatchInfoResponseItem(UpdateItem::Transaction((_seq, _digest))))) => {
-                            if !self.state.database.effects_exists(&_digest)? {
-                                queue.push(async move {
-                                    tokio::time::sleep(Duration::from_millis(EACH_ITEM_DELAY_MS)).await;
-                                    _digest
-                                });
+                        Some(Ok(BatchInfoResponseItem(UpdateItem::Batch(_signed_batch)) )) => {},
 
-                            }
+                        // Upon receiving a transaction digest, store it if it is not processed already.
+                        Some(Ok(BatchInfoResponseItem(UpdateItem::Transaction((seq, digest))))) => {
+                            if !self.state.database.effects_exists(&digest)? {
```

Comment:
Here we really have to wait for a little bit to not ask 4 times for something, and also have it already being processed, on each of the validators.